master/internal/trials/api_trial_source_info.go

```diff
@@ -0,0 +1,85 @@
+package trials
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/experiment"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/trialv1"
+)
+
+// TrialSourceInfoAPIServer is a dummy struct to do dependency injection to the api.
+// This allows us to define apiServer functions in sub-modules.
+type TrialSourceInfoAPIServer struct{}
+
+// ReportTrialSourceInfo creates a TrialSourceInfo, which serves as a link between
+// trials and checkpoints used for tracking purposes for fine tuning and inference.
+func (a *TrialSourceInfoAPIServer) ReportTrialSourceInfo(
+	ctx context.Context, req *apiv1.ReportTrialSourceInfoRequest,
+) (*apiv1.ReportTrialSourceInfoResponse, error) {
+	tsi := req.TrialSourceInfo
+	if err := CanGetTrialsExperimentAndCheckCanDoAction(ctx, int(tsi.TrialId),
+		experiment.AuthZProvider.Get().CanEditExperiment); err != nil {
+		return nil, err
```

Comment:
we want to double-check that this doesn't give away that a trial exists to a caller who shouldn't have view access. Have you tested that?

Reply:
applies anywhere else we're consulting our RBAC

Reply:
there is now testing for this in `TestTrialSourceInfoCheckpoint`